introduction_to_amazon_algorithms/linear_learner_abalone/Linear_Learner_Regression_csv_format.ipynb

```diff
@@ -438,37 +453,70 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "%%time\n",
-    "import json\n",
-    "from itertools import islice\n",
-    "import math\n",
-    "import struct\n",
+    "import io\n",
+    "import boto3\n",
+    "import random\n",
     "\n",
-    "#getting testing sample from our test file\n",
-    "test_data = [l for l in open(FILE_TEST, 'r')]\n",
-    "sample=random.choice(test_data).split(',')\n",
-    "actual_age=sample[0] \n",
-    "payload=sample[1:] #removing actual age from the sample\n",
-    "payload=','.join(map(str, payload))\n",
     "\n",
+    "def data_split(\n",
```

Comment:
This function is pretty complicated and could use refactoring. I can make some suggestions, but could we also use a built-in to a standard science library like numpy/pandas/scikit here instead?

Reply:
The function written is identical to what was written earlier. I just moved it to appendix. 

I would prefer not making changes at the last minute because testing might not be thorough. This can left for later especially because it is not part of the main body.